remote/remote.go

```diff
@@ -193,14 +193,13 @@ func newV1Image(keychain authn.Keychain, repoName string, platform imgutil.Platf
 		time.Sleep(100 * time.Duration(i) * time.Millisecond) // wait if retrying
 		image, err = remote.Image(ref, remote.WithAuth(auth), remote.WithTransport(http.DefaultTransport), remote.WithPlatform(v1Platform))
 		if err != nil {
-			if err == io.EOF && i != maxRetries {
+			transportErr, ok := err.(*transport.Error)
+			isValid40x := ok && is40x(transportErr)
+			if err == io.EOF || isValid40x && i != maxRetries {
```

Comment:
I think this has the subtle bug that @yaelharel found here: https://github.com/buildpacks/imgutil/pull/99/files (fixed in the above suggestion)

Reply:
You're so right!
We can move `i != maxRetries` to the beginning of the condition but I think that your suggestion above is better.

Reply:
@jjbustamante, just to make it clear, if we got an error after `maxRetries`, we want to return an error that is defined in the `for` loop and not get to line 211.

Reply:
Yeap, got it! Thanks!